eng/references.targets

```diff
@@ -37,15 +37,19 @@
 
   <!-- Make shared framework assemblies not app-local (non private). -->
   <Target Name="UpdateProjectReferencesWithPrivateAttribute"
-          AfterTargets="AssignProjectConfiguration"
-          BeforeTargets="PrepareProjectReferences"
           Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
-                     ('$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true') and
-                     '@(ProjectReferenceWithConfiguration)' != ''">
+                     ('$(IsSourceProject)' == 'true' or '$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true') and
+                     '@(ProjectReference)' != ''">
     <ItemGroup>
-      <ProjectReferenceWithConfiguration PrivateAssets="all"
-                                         Private="false"
-                                         Condition="$(NetCoreAppLibrary.Contains('%(Filename);'))" />
+      <_noPackageNetCoreAppLibrary Include="@(NetCoreAppLibrary)"
+                                   Condition="'%(NetCoreAppLibrary.IsPackable)' != 'true'" />
```

Comment:
If the only consumer of this metadata is as a list without IsPackable, perhaps we can just store it that way and avoid the metadata?  IOW we have `NoPackageNetCoreAppLibrary` item statically defined instead of calculated in this target.

Reply:
I prefer metadata over a separate list. Please check my latest changes in which I removed the target entirely in favor of static evaluation.